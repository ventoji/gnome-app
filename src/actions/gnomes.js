
export const addGnomes = (gnomes) => ({
    type: 'ADD_GNOMES',
    gnomes
  });