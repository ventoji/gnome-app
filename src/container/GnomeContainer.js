import {connect} from 'react-redux';
import Gnome from '../components/Gnome';
import { withRouter } from 'react-router-dom';

const mapStateToProps = (state) => {
    return {
        state: state.gnome
    };
};

const GnomeContainer = connect(mapStateToProps)(Gnome);

export default withRouter(GnomeContainer);

