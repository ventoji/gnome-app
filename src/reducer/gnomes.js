export default (state=[], action) => {

    switch (action.type){

        case 'ADD_GNOMES':
            return  action.gnomes;

        default:
            return state;

    }
};