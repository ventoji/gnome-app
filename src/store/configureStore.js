import {createStore} from 'redux';
import { persistStore, persistReducer } from 'redux-persist';
import storage from 'redux-persist/lib/storage';


import createRootReducer from '../reducer';


const persistConfig = {
    key: 'root',
    storage: storage,
    whitelist: ['gnome', 'gnomes']
    };

const pReducer = persistReducer(persistConfig, createRootReducer());


export const configureStore = createStore(
    pReducer
 );
 
 export const persistor = persistStore(configureStore);

/* export default () => {

    const store = createStore(
        combineReducers({
            gnome: gnomeReducer,
            filters: filtersReducer,
            gnomes: gnomesReducer
        })
     );

    return store;
}; */