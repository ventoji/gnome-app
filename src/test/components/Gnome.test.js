import React from 'react';
import {BrowserRouter } from 'react-router-dom';
import GnomeContainer from '../../container/GnomeContainer';
import {gnomeItem} from '../fixtures/gnomes';

import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';

const mockStore = configureStore([]);

describe('Connected Gnome React-Redux Component', () => {
    let store;
    let component;
    beforeEach(() => {
      store = mockStore({
        gnome: gnomeItem
      });
      component = renderer.create(
        <Provider store={store}>
            <BrowserRouter >
                <GnomeContainer />
            </BrowserRouter>
        </Provider>
      );
    });
    it('should render GnomeContainer properly ', () => {
        expect(component.toJSON()).toMatchSnapshot();
    });
  });


/* test('should render Gnome correctly', () => {
    expect(wrapper).toMatchSnapshot();
}); */