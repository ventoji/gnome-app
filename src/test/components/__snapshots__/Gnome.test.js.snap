// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Connected Gnome React-Redux Component should render GnomeContainer properly  1`] = `
<section
  className="gnome-content"
>
  <h1
    className="gnome__name"
  >
    Tobus Quickwhistle
     
  </h1>
  <div
    className="gnome"
  >
    <div
      className="gnome__photo"
    >
      <figure
        className="gnome__photo-container "
      >
        <img
          alt=" profile photo"
          className="gnome__img"
          src="http://www.publicdomainpictures.net/pictures/10000/nahled/thinking-monkey-11282237747K8xB.jpg"
        />
      </figure>
    </div>
    <div
      className="gnome__personal-info"
    >
      <ul>
        <li
          className="gnome__personal-info-item"
        >
           
          <span
            className="gnome__personal-info-title"
          >
             Age: 
          </span>
           
          <p
            className="gnome__personal-info-item"
          >
             
            306
          </p>
        </li>
        <li
          className="gnome__personal-info-item"
        >
           
          <span
            className="gnome__personal-info-title"
          >
             height: 
          </span>
           
          <p
            className="gnome__personal-info-item"
          >
             
            107.75835
          </p>
        </li>
        <li
          className="gnome__personal-info-item"
        >
           
          <span
            className="gnome__personal-info-title"
          >
             Weight: 
          </span>
           
          <p
            className="gnome__personal-info-item"
          >
             
            39.065952
          </p>
        </li>
        <li
          className="gnome__personal-info-item"
        >
           
          <span
            className="gnome__personal-info-title"
          >
             Hair color: 
          </span>
           
          <p
            className="gnome__personal-info-item"
          >
             
            Pink
          </p>
        </li>
      </ul>
      <div
        className="gnome__professions"
      >
        <span
          className="gnome__personal-info-title"
        >
           Professions: 
        </span>
        <p
          className="gnome__personal-info-item gnome__personal-info-item--profession"
        >
          Metalworker
           
        </p>
        <p
          className="gnome__personal-info-item gnome__personal-info-item--profession"
        >
          Woodcarver
           
        </p>
        <p
          className="gnome__personal-info-item gnome__personal-info-item--profession"
        >
          Stonecarver
           
        </p>
        <p
          className="gnome__personal-info-item gnome__personal-info-item--profession"
        >
           Tinker
           
        </p>
        <p
          className="gnome__personal-info-item gnome__personal-info-item--profession"
        >
          Tailor
           
        </p>
        <p
          className="gnome__personal-info-item gnome__personal-info-item--profession"
        >
          Potter
           
        </p>
      </div>
      <div
        className="gnome__friends"
      >
        <span
          className="gnome__personal-info-title"
        >
           Friends: 
        </span>
        <p
          className="gnome__personal-info-item gnome__personal-info-item--profession"
        >
          Cogwitz Chillwidget
           
        </p>
        <p
          className="gnome__personal-info-item gnome__personal-info-item--profession"
        >
          Tinadette Chillbuster
           
        </p>
      </div>
    </div>
  </div>
</section>
`;
